UI_DSM/UI_DSM.Server/Managers/RoleManager/RoleManager.cs

```diff
@@ -61,7 +86,7 @@ public async Task<IEnumerable<Role>> GetEntities()
         /// <returns>A <see cref="Task" /> with the result of the creation</returns>
         public async Task<EntityOperationResult<Role>> CreateEntity(Role entity)
         {
-            if (this.context.Roles.AsEnumerable().Any(x => x.RoleName.Equals(entity.RoleName, StringComparison.InvariantCultureIgnoreCase)))
+            if (this.context.UiDsmRoles.AsEnumerable().Any(x => x.RoleName.Equals(entity.RoleName, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
AsEnumerable downloads all UiDsmRoles from the server, before the Any command is executed. So execution of "Any()" is done client side. Is that what you want here?

Reply:
Unique constraint has been added : We can then remove that kind of check and just handling the exception correctly.